src/os/layer/preset/layerpresetmanager.js

```diff
@@ -1,223 +1,240 @@
+      const layerOptions = layer.getLayerOptions();
+      if (layerId && layerOptions && layerOptions['applyDefaultPresets']) {
+        // Apply default presets to the layer when configured.
+        this.getPresets(layerId, true);
+      }
     }
   }
-};
-
-
-/**
- * Gets a promise that resolves to the presets for a given layer type.
- *
- * @param {string} type
- * @param {string} url
- */
-os.layer.preset.LayerPresetManager.prototype.registerPreset = function(type, url) {
-  var promise = this.presets_[type];
-
-  if (!this.requested_[url]) {
-    this.requested_[url] = true;
-    var request = new os.net.Request(url);
-
-    if (promise) {
-      // presets are already loaded for that type, so attempt to load the new ones and replace the promise
-      promise.then(function(originalPresets) {
-        var newPromise = request.getPromise().then(function(result) {
-          var presets = /** @type {Array<osx.layer.Preset>} */ (JSON.parse(result));
-
-          if (presets && originalPresets) {
-            presets = presets.concat(originalPresets);
-          }
 
-          this.presets_[type] = newPromise;
-
-          return presets;
-        }, this.handleLoadError, this);
-      }, this.handleLoadError, this);
-    } else {
-      // no presets yet, so load them
-      this.presets_[type] = request.getPromise().then(function(result) {
-        var presets = /** @type {Array<osx.layer.Preset>} */ (JSON.parse(result));
-        return presets;
-      }, this.handleLoadError, this);
+  /**
+   * @param {boolean|undefined} b
+   * @return {!boolean}
+   */
+  isAdmin(b) {
+    if (b === true || b === false) {
+      this.admin_ = (b === true);
     }
+    return this.admin_;
   }
-};
 
+  /**
+   * @param {string} key
+   * @param {IPresetService} service
+   * @param {...} opt
+   * @return {boolean} true if prior value overwritten
+   */
+  registerService(key, service, ...opt) {
+    this.presets_ = {}; // set presets to reload every layer
+    return this.services_.register.apply(this.services_, [key, service].concat(opt));
+  }
 
-/**
- * Handler for errors in loading presets.
- *
- * @param {*} reason
- */
-os.layer.preset.LayerPresetManager.prototype.handleLoadError = function(reason) {
-  var msg = 'Unspecified error.';
-  if (typeof reason == 'string') {
-    msg = reason;
-  } else if (reason instanceof Error) {
-    msg = reason.message;
+  /**
+   * @param {string} key
+   * @return {boolean} true if prior value dropped
+   */
+  unregisterService(key) {
+    this.presets_ = {}; // set presets to reload every layer
+    return this.services_.remove(key);
   }
 
-  goog.log.error(os.layer.preset.LayerPresetManager.LOGGER_, 'Failed to load presets. Reason: ' + msg);
-};
+  /**
+   * Gets a promise that resolves to the presets for a given layer type.
+   *
+   * @param {string} type
+   * @param {string} url
+   * @deprecated Please create a os.layer.preset.PresetService instead
+   */
+  registerPreset(type, url) {}
```

Comment:
Are these here to avoid a breaking change? If so, I feel like they aren't really deprecated... it's still a breaking change because registering this way will still not work even if the code will still build. I'd rather we just remove them.